### PR TITLE
Adding our workflows

### PR DESCRIPTION
**Pull request by pantelis-karamolegkos**
[SRE-3593]

[SRE-3593]: https://workable.atlassian.net/browse/SRE-3593?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/ci_workable.yml [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+      - name: set the build branch
+        id: set-branch
+        shell: bash
+        run: |
+          if [[ "${{ inputs.manual_build_branch }}" == "" ]]; then
+            ACTUAL_BRANCH=${{ github.ref_name }}
+            echo "ref=${ACTUAL_BRANCH#workable_feature/*}" >> $GITHUB_OUTPUT
+          else
+            echo "ref=${{ inputs.manual_build_branch }}" >> $GITHUB_OUTPUT
+          fi
```
Comment (innnkkki) [inline .github/workflows/ci_workable.yml]: Just as a reminder to get the branch and tehn before sending it to $GITHUB_OUTPUT to replace workable_feature/ with ""

**Review by innnkkki — APPROVED** [review]

File: .github/workflows/ci_workable.yml (hunk 1 above)
Comment (pantelis-karamolegkos) [reply .github/workflows/ci_workable.yml]: Aren't I doing this?

```sh
▶ ACTUAL_BRANCH=workable_feature/lalaland

▶ ref=${ACTUAL_BRANCH#workable_feature/*}

▶ echo $ref
lalaland
```

**Review by pkossyfas — APPROVED** [review]

File: .github/workflows/ci_workable.yml [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+    if: github.repository == 'Workable/external-secrets'
+    environment: Workable
+    name: Build and push docker image
+    needs:
+      - set-ref
+      - set-tag
+    uses: Workable/external-secrets/.github/workflows/common_build_and_push_to_gcr.yaml@main
+    with:
+      dockerfile: Dockerfile_workable
+      build_branch: ${{ needs.set-ref.outputs.ref }}
```
Comment (innnkkki) [inline .github/workflows/ci_workable.yml]: I think that this will break with the change above. Should we do the change in line 220 instead?

Comment (pantelis-karamolegkos) [reply .github/workflows/ci_workable.yml]: I think the ouput name remains [ref](https://github.com/Workable/external-secrets/pull/3/files#diff-6c3e81e0026a0c0f4a80876122e2d4ce8bd8275037317bc61d23035d335fa0bcR209)

```
 echo "ref=${ACTUAL_BRANCH#workable_feature/*}" >> $GITHUB_OUTPUT
```

**Review by innnkkki — APPROVED** [review]

File: .github/workflows/ci_workable.yml (hunk 2 above)
Comment (pantelis-karamolegkos) [reply .github/workflows/ci_workable.yml]: addressed as discussed offline

**Pull request merged**